### PR TITLE
Add guide on contributing to the Biopython Tutorial

### PR DESCRIPTION
**Pull request by bganglia**
This pull request adds a guide explaining how to contribute to the Biopython tutorial.

It addresses [issue 156 in the biopython.github.io repository](https://github.com/biopython/biopython.github.io/issues/156)

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Doc/README.rst [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+Installing the tools you need
+-----------------------------
+
+On Ubuntu, you can install these with the following commands::
+
+    sudo apt install hevea
+    sudo apt install texlive-latex-base
+    sudo apt-get install texlive-latex-extra
+    sudo apt-get install python-pygments
```
Comment (peterjc) [inline Doc/README.rst]: Shouldn't all four lines be ``apt-get``? And can't that be one line?

```
sudo apt-get install hevea texlive-latex-base texlive-latex-extra python-pygments
```

File: Doc/README.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+Formatting code examples
+------------------------
+
+Code examples should be formatted using minted. Also, the Biopython tutorial uses its own system for testing code examples, which is located in ``test_Tutorial.py``. You can read more about this system in `section 21.4 of the Biopython Tutorial <http://biopython.org/DIST/docs/tutorial/Tutorial.html#htoc316>`_.
```
Comment (peterjc) [inline Doc/README.rst]: The exact section number will change, just say see the testing chapter in the Tutorial.

Comment (peterjc) [reply Doc/README.rst]: And for the same reason, avoid the HTML anchor, just link to the Tutorial.html itself.

File: Doc/README.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+Here is an example::
+
+    %doctest path-to-folder
+    \begin{minted}{python}
+    from Bio import SeqIO
+    \end{minted}
```
Comment (peterjc) [inline Doc/README.rst]: Should be:

```
    %doctest path-to-folder
    \begin{minted}{pycon}
    >>> from Bio import SeqIO
    \end{minted}
```

The ``{python}`` snippets without ``>>>`` are not doctests.

File: Doc/README.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    \label{chapter:chapterlabel}
+
+Section titles::
+
+    \section{Title of Section}
+    \label{sec:sectionlabel}
+
+Subsection titles::
+
+    \section{Title of Subsection}
```
Comment (peterjc) [inline Doc/README.rst]: Should be ``\subsection{Title of Subsection}``

Are you listing these in order to emphasise the label convention?

File: Doc/README.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+When referring to code in the middle of a paragraph, format it as follows::
+
+    \verb|ObjectName|
```
Comment (peterjc) [inline Doc/README.rst]: How about:

```
    \verb|variable_name|
```

and and then highlight in the comment below that this takes care of underscores etc.

File: Doc/README.rst [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+Build the documentation by running "make" in the Docs folder.
+
+Ideally, the documentation should be generated without a hitch, but some errors may cause the program to stop and ask for user input. Press R to continue building the documentation.
+
+If you would like to ignore errors at the command-line, then you can use::
+
+    yes R | make
+
+However, these errors may provide valuable information if there are mistakes in your own LaTeX code.
```
Comment (peterjc) [inline Doc/README.rst]: I would remove all of this general LaTeX advice about continuing with errors - if there are errors, they need fixing, so why encourage ignoring them?

File: Doc/README.rst (hunk 3 above)
Comment (peterjc) [reply Doc/README.rst]: Also ``\begin{minted}{python}`` (python code without ``>>>``) versus ``\begin{minted}{pycon}`` (python console with ``>>>``).

File: Doc/README.rst (hunk 1 above)
Comment (peterjc) [reply Doc/README.rst]: After a quick google, I see "apt" has replaced "apt-get" as the preferred command here. Anyway, one line seems clearer :)

File: Doc/README.rst (hunk 4 above)
Comment (bganglia) [reply Doc/README.rst]: @peterjc Yes, emphasizing the label convention was the idea. I am not sure if it is really necessary though.

File: Doc/README.rst (hunk 3 above)
Comment (bganglia) [reply Doc/README.rst]: I fixed the {pycon} example and added another example for {python}.

**Comment by peterjc** [comment]
There's still some trailing white space which the Travis tests are complaining about.

```
$ doc8 Doc/
Scanning...
Validating...
Doc/README.rst:15: D002 Trailing whitespace
Doc/README.rst:16: D002 Trailing whitespace
Doc/README.rst:58: D002 Trailing whitespace
Doc/README.rst:59: D002 Trailing whitespace
Doc/README.rst:67: D002 Trailing whitespace
```

Looks like ``CONTRIBUTING.rst`` was slightly out of date as we're actually using ``doc8`` as well as ``rst-lint`` here, https://pypi.org/project/doc8/ - fixed in b3160201c5e3e201a2c31771041f38e11e091fc7

Cross reference #2580 on pre-commit hooks to also cover the RST files.

**Comment by bganglia** [comment]
@peterjc I fixed the whitespace issue. What do you think?

**Comment by peterjc** [comment]
Excellent - thank you!

**Pull request merged**